### DISCUSSION TITLE
What's the difference between `createRouteHandlerClient()` and `createServerComponentClient()`?

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#15907] [Questions]
As the title states I'm asking myself what is the difference between `createRouteHandlerClient()` and `createServerComponentClient()`?
I know that the `createServerComponentClient()` is used in Next js in Server components and `createRouteHandlerClient()`  is used in `route.ts` files but I figured out both are working fine in `route.ts` files which is tempting to make mistakes. So I came to the question of how they differ from each other.
As I side note I would add it's a very misleading naming as it sounds like `createRouteHandlerClient()` (Client) is used in the front end but as far as I know it seems to be used only serverside right?

**Comment by j4w8n** [comment]
The route handler version can get, set, and delete cookies. The server component version can only get cookies.

The trailing "Client" wording is referring to the supabase client

**Reply by bennik88** [reply]
Thank you for an that answer. That helps to understand them better and makes it more clear for me! I Hope the docs will have more details about that soon.
